Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,387 @@
+        # get the edge point where we will rotate t0 around
+        square_diagonal_length = math.sqrt(2 * base_length ** 2)
+        edge = self._get_point_on_square_edge(
+            base_center, square_diagonal_length, base_angle - math.pi / 4)
+
+        # translate edge to origin, temp point is is copy of edge
+        temp = Point(edge.x - t0.x, edge.y - t0.y)
+        # rotate point around edge point and translate back
+        t1 = Point(
+            temp.x * math.cos(alpha) - temp.y * math.sin(alpha),
+            temp.x * math.sin(alpha) + temp.y * math.cos(alpha)
+        )
+        # translate rotated point back to its original spot
+        t1 = Point(t1.x + t0.x, t1.y + t0.y)
+
+        # calculate the middle point between the rotated point and edge
+        t2 = Point((t1.x + edge.x) / 2, (t1.y + edge.y) / 2)
+        return self._get_point_on_square_edge(t2, length, angle)
+
+    @staticmethod
+    def _get_point_on_square_edge(center, length, angle):
+        """Calculate the central point on the drawing edge of the given square.
+
+        Parameters
+        ----------
+        center : Point
+            The square center point.
+        length : float
+            The square side length.
+        angle : float
+            The angle of the square.
+
+        Returns
+        -------
+        Point
+            A point on the center of the drawing edge of the given square.
+
+        """
+        # by default, we draw on top, so we need to offset the angle by 90degs
+        rotate_90 = math.pi / 2
+        return Point(
+            center.x + length / 2 * math.cos(angle + rotate_90),
+            center.y + length / 2 * math.sin(angle + rotate_90)
+        )
+
+
+class SklTreeAdapter:
+    def __init__(self, tree):
+        self._tree = tree
+
+    def weight(self, node):
+        return self.num_samples(node) / self.num_samples(self.parent(node))
+
+    def num_samples(self, node):
+        return self._tree.n_node_samples[node]
+
+    def parent(self, node):
+        for idx, el in enumerate(self._tree.children_left):
+            if el == node:
+                return idx
+        for idx, el in enumerate(self._tree.children_right):
+            if el == node:
+                return idx
+        return -1
+
+    def has_children(self, node):
+        return self._tree.children_left[node] != -1 \
+               or self._tree.children_right[node] != -1
+
+    def left_child(self, node):
+        return self._tree.children_left[node]
+
+    def right_child(self, node):
+        return self._tree.children_right[node]
+
+
+if __name__ == "__main__":
+    import sys
+    from Orange.classification.tree import TreeLearner
+
+    argv = sys.argv
+    if len(argv) > 1:
+        filename = argv[1]
+    else:
+        filename = "iris"
+
+    app = QtGui.QApplication(argv)
+    ow = OWPythagorasTree()
+    data = Orange.data.Table(filename)
+    clf = TreeLearner(max_depth=10)(data)
```

Comment:
@astaric, @zidarsk8 What do we do with pylint complaining about these semantically local variables [interpreted as constants](https://travis-ci.org/biolab/orange3/jobs/120147877)?